### PR TITLE
Several Pylint suggestions

### PR DESCRIPTION
**Pull request by cmc333333**
Using pylint as a guide, I went through and fixed up lots of little odds and ends. For the most part, these were little things like indentation, variable names overriding functions, unused arguments, etc., but there's a handful of more important changes. Each of the `[]` default args was a potential bug; in one case we were testing the presence of a generator (always true); in a third, the SxS command hadn't been updated (pylint pointed out that we were using a non-iterable object as an iterable). There's a lot here, but it's all low risk. Easiest to review by changeset.

### CODE REVIEW

File: regparser/layer/def_finders.py [hunk 1]
```diff
@@ -122,8 +122,8 @@ def find(self, node):
         refs = []
         tagged_text = getattr(node, 'tagged_text', '')
         for match, _, _ in grammar.xml_term_parser.scanString(tagged_text):
-            """Position in match reflects XML tags, so its dropped in
-            preference of new values based on node.text."""
+            # Position in match reflects XML tags, so its dropped in
+            # preference of new values based on node.text."""
```
Comment (tadhg-ohiggins) [inline regparser/layer/def_finders.py]: Very minor, but we probably want to lose the trailing `"""` here.

File: regparser/layer/formatting.py [hunk 2]
```diff
@@ -236,7 +236,7 @@ def match_data(self, match):
 
 
 class Footnotes(PlaintextFormatData):
-    """E.g.     [^4](Contents of footnote with escaped parens: \(\)s)"""
+    r"""E.g.     [^4](Contents of footnote with escaped parens: \(\)s)"""
```
Comment (tadhg-ohiggins) [inline regparser/layer/formatting.py]: ? Do we want to mark this comment as regex text?

Comment (cmc333333) [inline regparser/layer/formatting.py]: The `r` prefix just means "raw", so we don't need to double-escape the slashes. If we don't use an `r`, we're technically encoding bad escape sequences (`\(` and `\)`). What should we do? I'd like to include the indication that parens can be escaped, but not if it makes the code harder to understand.

Comment (tadhg-ohiggins) [inline regparser/layer/formatting.py]: I might not be understanding this correctly, but I would consider trying something like:

```
"""
Handles footnotes. A variable with text that would match looks something like this::

    fn = r"[^4](Contents of footnote with escaped parens: \(\)s)"
"""
```

**Pull request merged**